webapp/src/Controller/API/JudgehostController.php

```diff
@@ -550,6 +553,58 @@ public function addDebugInfo(
         $this->em->flush();
     }
 
+    /**
+     * Add visual team output.
+     */
+    #[IsGranted('ROLE_JUDGEHOST')]
+    #[Rest\Post('/add-visual/{hostname}/{judgeTaskId<\d+>}')]
+    #[OA\Response(response: 200, description: 'When the visual output has been added')]
+    public function addVisualization(
+        Request $request,
+        #[OA\PathParameter(description: 'The hostname of the judgehost that wants to add the debug info')]
+        string $hostname,
+        #[OA\PathParameter(description: 'The ID of the judgetask to add', schema: new OA\Schema(type: 'integer'))]
+        int $judgeTaskId
+    ): void {
+        $judgeTask = $this->em->getRepository(JudgeTask::class)->find($judgeTaskId);
+        if ($judgeTask === null) {
+            throw new BadRequestHttpException(
+                'Inconsistent data, no judgetask known with judgetaskid = ' . $judgeTaskId . '.');
+        }
+
+        foreach (['visual_output', 'testcase_id'] as $argument) {
+            if (!$request->request->has($argument)) {
+                throw new BadRequestHttpException(
+                    sprintf("Argument '%s' is mandatory", $argument));
+            }
+        }
+
+        $judgehost = $this->em->getRepository(Judgehost::class)->findOneBy(['hostname' => $hostname]);
+        if (!$judgehost) {
+            throw new BadRequestHttpException("Who are you and why are you sending us any data?");
+        }
+
+        $judging = $this->em->getRepository(Judging::class)->find($judgeTask->getJobId());
+        if ($judging === null) {
+            throw new BadRequestHttpException(
+                'Inconsistent data, no judging known with judgingid = ' . $judgeTask->getJobId() . '.');
+        }
+        if ($tempFilename = tempnam($this->dj->getDomjudgeTmpDir(), "visual-")) {
```

Comment:
This means it doesn't work with 2 servers in HA mode. We should store this in the DB as a file.

Reply:
This is loosely based on what we do with debug packages, so we have the same problem there.

Reply:
Ah I didn't know. Then maybe for now that's fine but we should fix it at some point?